contracts/examples/NFTExample.sol

```diff
@@ -38,29 +35,17 @@ contract NFTExample is
             super.supportsInterface(interfaceId);
     }
 
-    function setTokenURI(
-        uint256 tokenId,
-        ctUint64 _itTokenURI,
-        bytes calldata _itSignature
-    ) public {
-        _requireOwned(tokenId);
-        address owner = _ownerOf(tokenId);
-        if (msg.sender != owner) {
-            revert ERC721IncorrectOwner(msg.sender, tokenId, owner);
-        }
-
-        itUint64 memory it;
-        it.ciphertext = _itTokenURI;
-        it.signature = _itSignature;
-
-        _setTokenURI(owner, tokenId, MpcCore.validateCiphertext(it));
-    }
-
-    function mint(address to) public onlyOwner {
+    function mint(
+        address to,
+        ctUint64[] calldata _itTokenURI,
+        bytes[] calldata _itSignature
+    ) public onlyOwner {
         uint256 tokenId = _totalSupply;
```

Comment:
move lines 43-45 including 49 to base class

Reply:
total supply is not part of the ERC721 standard so maybe we should keep it as is for now. Additionally, we might want to consider swapping the confidential ERC721 for the OpenZeppelin ERC721 since we have removed all the "confidential" logic from it